File: Đồ án OOP C# - Quản lý phòng bán vé xe khách/DoAn2/LaiXe.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoAn2
{
    public class LaiXe : NhanVien
    {
        private int dPhucap;
        private List<LaiXe> ldslaixe = new List<LaiXe>();
        //
        public int Phucap
        {
            get { return this.dPhucap; }
            set { this.dPhucap = value; }
        }
        public List<LaiXe> dslaixe
        {
            get { return this.ldslaixe; }
            set { this.ldslaixe = value; }
        }
        public LaiXe() : base()
        {

        }
        //
        public new void Nhap()
        {
            base.Nhap();
            Console.Write("Nhap phu cap: ");
            this.dPhucap = int.Parse(Console.ReadLine());
            this.iLuong = tinhluong();
        }
        //
        public override void Xuat()
        {
            Console.WriteLine("     -----------------------------------------------------DANH SÁCH NHÂN VIÊN LÁI XE-------------------------------------------------------");
            Console.WriteLine("     ______________________________________________________________________________________________________________________________________");
            Console.Write(string.Format("{0,8 } |", "STT"));
            Console.Write(string.Format(" {0, -12} |", "Mã nhân viên"));
            Console.Write(string.Format(" {0, -19} |", "Họ và tên "));
            Console.Write(string.Format(" {0, -15} |", "Số CMND"));
            Console.Write(string.Format(" {0, -12} |", "Năm sinh"));
            Console.Write(string.Format(" {0, -12} |", "Ngày công"));
            Console.Write(string.Format(" {0, -19} |", "Lương cơ bản"));
            Console.WriteLine(string.Format(" {0, -20} |", "Lương chính thức" + " $"));
            Console.WriteLine("    --------------------------------------------------------------------------------------------------------------------------------------");
            for (int i = 0; i < dslaixe.Count; i++)
            {
                Console.Write(string.Format("{0, 8} |", i + 1));
                Console.Write(string.Format(" {0, -12} |", dslaixe[i].iManv));
                Console.Write(string.Format(" {0, -19} |", dslaixe[i].sHoten));
                Console.Write(string.Format(" {0, -15} |", dslaixe[i].sCMND));
                Console.Write(string.Format(" {0, -12} |", dslaixe[i].iNamsinh));
                Console.Write(string.Format(" {0, -12} |", dslaixe[i].dNgaycong));
                Console.Write(string.Format(" {0, -19} |", dslaixe[i].iLuongcb));
                Console.WriteLine(string.Format(" {0, -20} |", dslaixe[i].iLuong + " $"));
                Console.WriteLine("    ------------------------------------------------------------------------------------------------------------------------------------------");
            }
        }
        //
        public override int tinhluong()
        {
            this.iLuong = this.dNgaycong * this.Phucap + this.iLuong;
            return this.iLuong;
        }
        public override void GhifileNv(String filename)
        {
            FileStream fnv = new FileStream(filename, FileMode.Create);
            StreamWriter swrite = new StreamWriter(fnv, Encoding.UTF8);
            for (int i = 0; i < dslaixe.Count; i++)
            {
                swrite.WriteLine(dslaixe[i].Manv);
                swrite.WriteLine(dslaixe[i].Hoten);
                swrite.WriteLine(dslaixe[i].CMND);
                swrite.WriteLine(dslaixe[i].Namsinh);
                swrite.WriteLine(dslaixe[i].Ngaycong);
                swrite.WriteLine(dslaixe[i].Luongcb);
                swrite.WriteLine(dslaixe[i].Phucap);
                swrite.WriteLine(dslaixe[i].Luong);
            }
            swrite.Flush();
            dslaixe = new List<LaiXe>();
            fnv.Close();
        }
        public override void DocfileNv(String filename)
        {
            FileStream fp = new FileStream(filename, FileMode.Open);
            StreamReader rd = new StreamReader(fp, Encoding.UTF8);
            int dem = 0;
            while (rd.ReadLine() != null)
            {
                dem++;
            }
            int size = dem / 8;
            rd.Close();
            fp.Close();
            FileStream fp1 = new FileStream(filename, FileMode.Open);
            StreamReader rd1 = new StreamReader(fp1, Encoding.UTF8);
            for (int i = 0; i < size; i++)
            {
                LaiXe lx = new LaiXe();
                lx.Manv = int.Parse(rd1.ReadLine().Replace("\\uFEFF", ""));
                lx.Hoten = rd1.ReadLine();
                lx.CMND = int.Parse(rd1.ReadLine().Replace("\\uFEFF", ""));
                lx.Namsinh = int.Parse(rd1.ReadLine().Replace("\\uFEFF", ""));
                lx.Ngaycong = int.Parse(rd1.ReadLine().Replace("\\uFEFF", ""));
                lx.Luongcb = int.Parse(rd1.ReadLine().Replace("\\uFEFF", ""));
                lx.Phucap = int.Parse(rd1.ReadLine().Replace("\\uFEFF", ""));
                lx.Luong = int.Parse(rd1.ReadLine().Replace("\\uFEFF", ""));
                dslaixe.Add(lx);
            }
            fp1.Close();
            rd1.Close();
        }
        public override void Them()
        {
            int n;
            Console.Write("So nhan vien lai xe muon nhap: ");
            n = Convert.ToInt32(Console.ReadLine());
            for (int i = 0; i < n; i++)
            {
                LaiXe lx = new LaiXe();
                lx.Nhap();
                dslaixe.Add(lx);
            }
        }
        public override void Xoa(int temp)
        {
            List<LaiXe> tam = dslaixe;
            dslaixe = new List<LaiXe>();
            temp = temp - 1;
            // Neu temp <= 0 => Xoa dau
            if (temp < 0)
            {
                temp = 0;
            }
            // Neu temp >= n => Xoa cuoi
            else if (temp >= tam.Count)
            {
                temp = tam.Count - 1;
            }
            // Dich chuyen mang
            for (int i = 0; i < tam.Count; i++)
            {
                LaiXe TT = new LaiXe();
                if (i != temp)
                {
                    TT = tam[i];
                    dslaixe.Add(TT);
                }
            }
        }
        public override void Sua(int temp)
        {
            temp = temp - 1;
            LaiXe lxmoi = new LaiXe();
            lxmoi.Nhap();
            dslaixe[temp] = lxmoi;
        }
        public static bool operator >(LaiXe a, LaiXe b)
        {
            if (a.iLuong < b.iLuong) return false;
            return true;
        }
        public static bool operator <(LaiXe a, LaiXe b)
        {
            if (a.iLuong > b.iLuong) return false;
            return true;
        }
        public  void sosanh2()
        {
            int dem = 0;
            int vt1 = 0;
            int vt2 = 0;
            Console.WriteLine("Nhập nhân viên thứ nhất cần tìm: ");
            int NV1 = int.Parse(Console.ReadLine());
            Console.WriteLine("Nhập nhân viên thứ hai cần tìm: ");
            int NV2 = int.Parse(Console.ReadLine());
            for (int i = 0; i < dslaixe.Count; i++)
            {
                if (dslaixe[i].Manv == NV1)
                {
                    dem++;
                    vt1 = i;
                }
                if (dslaixe[i].Manv == NV2)
                {
                    dem++;
                    vt2 = i;
                }
            }
            if (dem <= 1) Console.WriteLine("---> không tìm thấy <--- ");
            if (dem > 1)
            {
                if (dslaixe[vt1] > dslaixe[vt2]) Console.WriteLine("Chuyến có mã " + dslaixe[vt1].Hoten + " lớn hơn " + dslaixe[vt2].Hoten);
                if (dslaixe[vt1] < dslaixe[vt2]) Console.WriteLine("Chuyến có mã " + dslaixe[vt1].Hoten + " bé hơn " + dslaixe[vt2].Hoten);
                if (dslaixe[vt1] == dslaixe[vt2]) Console.WriteLine("Chuyến có mã " + dslaixe[vt1].Hoten + " bằng " + dslaixe[vt2].Hoten);
               
            }
        }
        public override void sort()
        {
            LaiXe t;
            for (int i = 0; i < dslaixe.Count - 1; i++)
            {
                for (int j = i + 1; j < dslaixe.Count; j++)
                {
                    if (dslaixe[i] > dslaixe[j])
                    {
                        t = dslaixe[i];
                        dslaixe[i] = dslaixe[j];
                        dslaixe[j] = t;
                    }
                }
            }
            GhifileNv("dslaixe.txt");
            dslaixe = new List<LaiXe>();
        }
        public override void search()
        {
            int dem1 = 0;
            Console.WriteLine("Nhập họ và tên nhân viên lái xe cần tìm: ");
            string NV = Console.ReadLine();
            Console.WriteLine("     _________________________________________________________________________________________________________________________________________");
            Console.Write(string.Format("{0,8 } |", "STT"));
            Console.Write(string.Format(" {0, -12} |", "Mã nhân viên"));
            Console.Write(string.Format(" {0, -19} |", "Họ và tên "));
            Console.Write(string.Format(" {0, -15} |", "Số CMND"));
            Console.Write(string.Format(" {0, -12} |", "Năm sinh"));
            Console.Write(string.Format(" {0, -12} |", "Ngày công"));
            Console.Write(string.Format(" {0, -19} |", "Lương cơ bản"));
            Console.WriteLine(string.Format(" {0, -20} |", "Lương chính thức" + " $"));
            Console.WriteLine("    ------------------------------------------------------------------------------------------------------------------------------------------");
            for (int i = 0; i < dslaixe.Count; i++)
            {
                if (dslaixe[i].sHoten == NV)
                {
                    dem1++;
                    Console.Write(string.Format("{0, 8} |", i + 1));
                    Console.Write(string.Format(" {0, -12} |", dslaixe[i].iManv));
                    Console.Write(string.Format(" {0, -19} |", dslaixe[i].sHoten));
                    Console.Write(string.Format(" {0, -15} |", dslaixe[i].sCMND));
                    Console.Write(string.Format(" {0, -12} |", dslaixe[i].iNamsinh));
                    Console.Write(string.Format(" {0, -12} |", dslaixe[i].dNgaycong));
                    Console.Write(string.Format(" {0, -19} |", dslaixe[i].iLuongcb));
                    Console.WriteLine(string.Format(" {0, -20} |", dslaixe[i].iLuong + " $"));
                    Console.WriteLine("   ----------------------------------------------------------------------------------------------------------------------------------");
                }
            }
                if (dem1 == 0) Console.WriteLine("---> không tìm thấy <--- ");
            
        }
        public override void luongmax()
        {
            Console.WriteLine("     _____________________________________________________NHÂN VIÊN CÓ LƯƠNG CAO NHẤT_______________________________________________________");

            Console.Write(string.Format("{0,8 } |", "STT"));
            Console.Write(string.Format(" {0, -12} |", "Mã nhân viên"));
            Console.Write(string.Format(" {0, -19} |", "Họ và tên "));
            Console.Write(string.Format(" {0, -15} |", "Số CMND"));
            Console.Write(string.Format(" {0, -12} |", "Năm sinh"));
            Console.Write(string.Format(" {0, -12} |", "Ngày công"));
            Console.Write(string.Format(" {0, -19} |", "Lương cơ bản"));
            Console.WriteLine(string.Format(" {0, -20} |", "Lương chính thức" + " $"));
            Console.WriteLine("    ----------------------------------------------------------------------------------------------------------------------------------------");
            double maxve = 0;
            int vt = 0;
            for (int i = 0; i < dslaixe.Count; i++)
            {
                if (dslaixe[i].tinhluong() > maxve)
                {
                    maxve = dslaixe[i].tinhluong();
                    vt = i;
                }
            }
            Console.Write(string.Format("{0, 8} |", vt + 1));
            Console.Write(string.Format(" {0, -12} |", dslaixe[vt].iManv));
            Console.Write(string.Format(" {0, -19} |", dslaixe[vt].sHoten));
            Console.Write(string.Format(" {0, -15} |", dslaixe[vt].sCMND));
            Console.Write(string.Format(" {0, -12} |", dslaixe[vt].iNamsinh));
            Console.Write(string.Format(" {0, -12} |", dslaixe[vt].dNgaycong));
            Console.Write(string.Format(" {0, -19} |", dslaixe[vt].iLuongcb));
            Console.WriteLine(string.Format(" {0, -20} |", dslaixe[vt].iLuong + " $"));
            Console.WriteLine("    ----------------------------------------------------------------------------------------------------------------------------------------");
        }

        public override void luongmin()
        {
            Console.WriteLine("     ______________________________________________________NHÂN VIÊN CÓ LƯƠNG THẤP NHẤT_____________________________________________________");

            Console.Write(string.Format("{0,8 } |", "STT"));
            Console.Write(string.Format(" {0, -12} |", "Mã nhân viên"));
            Console.Write(string.Format(" {0, -19} |", "Họ và tên "));
            Console.Write(string.Format(" {0, -15} |", "Số CMND"));
            Console.Write(string.Format(" {0, -12} |", "Năm sinh"));
            Console.Write(string.Format(" {0, -12} |", "Ngày công"));
            Console.Write(string.Format(" {0, -19} |", "Lương cơ bản"));
            Console.WriteLine(string.Format(" {0, -20} |", "Lương chính thức" + " $"));
            Console.WriteLine("    ---------------------------------------------------------------------------------------------------------------------------------------");
            double maxve = dslaixe[0].tinhluong();
            int vt = 0;
            for (int i = 0; i < dslaixe.Count; i++)
            {
                if (dslaixe[i].tinhluong() < maxve)
                {
                    maxve = dslaixe[i].tinhluong();
                    vt = i;
                }
            }
            Console.Write(string.Format("{0, 8} |", vt + 1));
            Console.Write(string.Format(" {0, -12} |", dslaixe[vt].iManv));
            Console.Write(string.Format(" {0, -19} |", dslaixe[vt].sHoten));
            Console.Write(string.Format(" {0, -15} |", dslaixe[vt].sCMND));
            Console.Write(string.Format(" {0, -12} |", dslaixe[vt].iNamsinh));
            Console.Write(string.Format(" {0, -12} |", dslaixe[vt].dNgaycong));
            Console.Write(string.Format(" {0, -19} |", dslaixe[vt].iLuongcb));
            Console.WriteLine(string.Format(" {0, -20} |", dslaixe[vt].iLuong + " $"));
            Console.WriteLine("    --------------------------------------------------------------------------------------------------------------------------------------");

        }
    }
}

File: Đồ án OOP Java - Quản lý phòng vé bán xe khách/src/Doan2/LaiXe.java
package Doan2;

import java.util.*;


import java.io.*;

public class LaiXe extends Nhanvien
{
	protected int iPhucap;
	protected ArrayList<LaiXe> dslaixe = new ArrayList<LaiXe>();
	BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
	//
	public int getPhucap()
	{
		return this.iPhucap;
	}
	public void setPhucap(int value)
	{
		this.iPhucap = value;
	}
	public ArrayList<LaiXe> getdslaixe()
	{
		return this.dslaixe;
	}
	public void setDslaixe(ArrayList<LaiXe> value)
	{
		this.dslaixe = value;
	}
	public LaiXe()
	{
		super();

	}
	//
	public void Nhap() throws IOException
	{
		super.Nhap();
		System.out.print("Nhập phụ cấp: ");
		this.iPhucap = Integer.parseInt(input.readLine());
		this.iLuong = tinhluong();
	}
	//
	@Override
	public void Xuat()
	{
		System.out.println("     -----------------------------------------------------DANH SÁCH NHÂN VIÊN LÁI XE-------------------------------------------------------");
		System.out.println("     ______________________________________________________________________________________________________________________________________");
		System.out.print(String.format("%1$8s |", "STT"));
		System.out.print(String.format(" %1$-12s |", "Mã nhân viên"));
		System.out.print(String.format(" %1$-19s |", "Họ và tên "));
		System.out.print(String.format(" %1$-15s |", "Số CMND"));
		System.out.print(String.format(" %1$-12s |", "Năm sinh"));
		System.out.print(String.format(" %1$-12s |", "Ngày công"));
		System.out.print(String.format(" %1$-19s |", "Lương cơ bản"));
		System.out.println(String.format(" %1$-20s |", "Lương chính thức" + " $"));
		System.out.println("    --------------------------------------------------------------------------------------------------------------------------------------");
		for (int i = 0; i < dslaixe.size(); i++)
		{
			System.out.print(String.format("%1$8s |", i + 1));
			System.out.print(String.format(" %1$-12s |", dslaixe.get(i).iManv));
			System.out.print(String.format(" %1$-19s |", dslaixe.get(i).sHoten));
			System.out.print(String.format(" %1$-15s |", dslaixe.get(i).iCMND));
			System.out.print(String.format(" %1$-12s |", dslaixe.get(i).iNamsinh));
			System.out.print(String.format(" %1$-12s |", dslaixe.get(i).iNgaycong));
			System.out.print(String.format(" %1$-19s |", dslaixe.get(i).iLuongcb));
			System.out.println(String.format(" %1$-20s |", dslaixe.get(i).iLuong + " $"));
			System.out.println("    ------------------------------------------------------------------------------------------------------------------------------------------");
		}
	}
	//
	@Override
	public int tinhluong()
	{
		this.iLuong = this.iNgaycong * this.getPhucap() + this.iLuong;
		return this.iLuong;
	}
	@Override
	public void GhifileNv(String filename) throws IOException
	{
		BufferedWriter pw = new BufferedWriter(new FileWriter(filename)) ;
		for(int i=0;i<dslaixe.size();i++)
		{
			pw.write(dslaixe.get(i).getManv()+ "\n");
			pw.write(dslaixe.get(i).getHoten() + "\n");
			pw.write(dslaixe.get(i).getCMND()+ "\n");
			pw.write(dslaixe.get(i).getNamsinh()+ "\n");
			pw.write(dslaixe.get(i).getNgaycong()+ "\n");
			pw.write(dslaixe.get(i).getLuongcb()+ "\n");
			pw.write(dslaixe.get(i).getPhucap()+ "\n");
			pw.write(dslaixe.get(i).getLuong()+ "\n");
		}
		pw.close();
		dslaixe.clear();	
	}
	@Override
	public void DocfileNv(String filename) throws IOException {
		// TODO Auto-generated method stub
		FileReader fp = new FileReader(filename);
		BufferedReader rd = new BufferedReader(fp);
		int dem = 0;
		while (rd.readLine() != null)
		{
			dem++;
		}
		int size = dem / 8;
		rd.close();
		fp.close();
		FileReader fp1 = new FileReader(filename);
		BufferedReader rd1 = new BufferedReader(fp1);
		for (int i = 0; i < size; i++)
		{
			LaiXe lx = new LaiXe();
			lx.setManv(Integer.parseInt(rd1.readLine().replaceAll("\\uFEFF", "")));
			lx.setHoten(rd1.readLine());
			lx.setCMND(Integer.parseInt(rd1.readLine().replaceAll("\\uFEFF", "")));
			lx.setNamsinh(Integer.parseInt(rd1.readLine().replaceAll("\\uFEFF", "")));
			lx.setNgaycong(Integer.parseInt(rd1.readLine().replaceAll("\\uFEFF", "")));
			lx.setLuongcb(Integer.parseInt(rd1.readLine().replaceAll("\\uFEFF", "")));
			lx.setPhucap(Integer.parseInt(rd1.readLine().replaceAll("\\uFEFF", "")));
			lx.setLuong(Integer.parseInt(rd1.readLine().replaceAll("\\uFEFF", "")));
			dslaixe.add(lx);
		}
		fp1.close();
		rd1.close();

	}
	@Override
	public void Them() throws IOException {
		// TODO Auto-generated method stub
		int n;
		System.out.print("Số nhân viên lái xe muốn nhập: ");
		n = Integer.parseInt(input.readLine());
		for (int i = 0; i < n; i++)
		{
			LaiXe lx = new LaiXe();
			lx.Nhap();
			dslaixe.add(lx);
		}

	}
	@Override
	public void Xoa(int temp) {
		// TODO Auto-generated method stub
		ArrayList<LaiXe> tm = new ArrayList<LaiXe>();
		ArrayList<LaiXe> tam = new ArrayList<LaiXe>();
		tm = dslaixe;
		dslaixe = tam;
		tam = tm;
		temp = temp - 1;
		// Neu temp <= 0 => Xoa dau
		if (temp < 0)
		{
			temp = 0;
		}
		// Neu temp >= n => Xoa cuoi
		else if (temp >= tam.size())
		{
			temp = tam.size() - 1;
		}
		// Dich chuyen mang
		for (int i = 0; i < tam.size(); i++)
		{
			LaiXe TT = new LaiXe();
			if (i != temp)
			{
				TT = tam.get(i);
				dslaixe.add(TT);
			}
		}

	}
	@Override
	public void Sua(int temp) throws IOException {
		// TODO Auto-generated method stub
		temp = temp - 1;
		LaiXe lxmoi = new LaiXe();
		lxmoi.Nhap();
		dslaixe.set(temp, lxmoi);

	}
	public static boolean opGreaterThan(LaiXe a, LaiXe b)
	{
			if (a.iLuong < b.iLuong)
				{
					return false;
				}
				return true;
	}
	public static boolean opLessThan(LaiXe a, LaiXe b)
	{
			if (a.iLuong > b.iLuong)
				{
					return false;
				}
				return true;
	}

	@Override
	public void sort() throws IOException {
		// TODO Auto-generated method stub
		LaiXe t;
		for (int i = 0; i < dslaixe.size() - 1; i++)
		{
			for (int j = i + 1; j < dslaixe.size(); j++)
			{
				if (opGreaterThan(dslaixe.get(i), dslaixe.get(j)))
				{
					t = dslaixe.get(i);
					dslaixe.set(i, dslaixe.get(j));
					//dslaixe.get(i).equals(dslaixe.get(j));
					dslaixe.set(j, t);
				}
			}
		}
		GhifileNv("dslaixe.txt");
		dslaixe.clear();
	}
	@Override
	public void search() throws IOException {
		// TODO Auto-generated method stub
		int dem1 = 0;
		System.out.println("Nhập họ và tên nhân viên lái xe cần tìm: ");
		String NV = input.readLine();
		System.out.println("     _________________________________________________________________________________________________________________________________________");
		System.out.print(String.format("%1$8s |", "STT"));
		System.out.print(String.format(" %1$-12s |", "Mã nhân viên"));
		System.out.print(String.format(" %1$-19s |", "Họ và tên "));
		System.out.print(String.format(" %1$-15s |", "Số CMND"));
		System.out.print(String.format(" %1$-12s |", "Năm sinh"));
		System.out.print(String.format(" %1$-12s |", "Ngày công"));
		System.out.print(String.format(" %1$-19s |", "Lương cơ bản"));
		System.out.println(String.format(" %1$-20s |", "Lương chính thức" + " $"));
		System.out.println("    ------------------------------------------------------------------------------------------------------------------------------------------");
		for (int i = 0; i < dslaixe.size(); i++)
		{
			if (NV.equals(dslaixe.get(i).sHoten))
			{
				dem1++;
				System.out.print(String.format("%1$8s |", i + 1));
				System.out.print(String.format(" %1$-12s |", dslaixe.get(i).iManv));
				System.out.print(String.format(" %1$-19s |", dslaixe.get(i).sHoten));
				System.out.print(String.format(" %1$-15s |", dslaixe.get(i).iCMND));
				System.out.print(String.format(" %1$-12s |", dslaixe.get(i).iNamsinh));
				System.out.print(String.format(" %1$-12s |", dslaixe.get(i).iNgaycong));
				System.out.print(String.format(" %1$-19s |", dslaixe.get(i).iLuongcb));
				System.out.println(String.format(" %1$-20s |", dslaixe.get(i).iLuong + " $"));
				System.out.println("   ----------------------------------------------------------------------------------------------------------------------------------");
			}
		}
			if (dem1 == 0)
			{
				System.out.println("---> không tìm thấy <--- ");
			}

	}
	@Override
	public void luongmax() {
		// TODO Auto-generated method stub
		System.out.println("     _____________________________________________________NHÂN VIÊN CÓ LƯƠNG CAO NHẤT_______________________________________________________");

		System.out.print(String.format("%1$8s |", "STT"));
		System.out.print(String.format(" %1$-12s |", "Mã nhân viên"));
		System.out.print(String.format(" %1$-19s |", "Họ và tên "));
		System.out.print(String.format(" %1$-15s |", "Số CMND"));
		System.out.print(String.format(" %1$-12s |", "Năm sinh"));
		System.out.print(String.format(" %1$-12s |", "Ngày công"));
		System.out.print(String.format(" %1$-19s |", "Lương cơ bản"));
		System.out.println(String.format(" %1$-20s |", "Lương chính thức" + " $"));
		System.out.println("    ----------------------------------------------------------------------------------------------------------------------------------------");
		double maxve = 0;
		int vt = 0;
		for (int i = 0; i < dslaixe.size(); i++)
		{
			if (dslaixe.get(i).tinhluong() > maxve)
			{
				maxve = dslaixe.get(i).tinhluong();
				vt = i;
			}
		}
		System.out.print(String.format("%1$8s |", vt + 1));
		System.out.print(String.format(" %1$-12s |", dslaixe.get(vt).iManv));
		System.out.print(String.format(" %1$-19s |", dslaixe.get(vt).sHoten));
		System.out.print(String.format(" %1$-15s |", dslaixe.get(vt).iCMND));
		System.out.print(String.format(" %1$-12s |", dslaixe.get(vt).iNamsinh));
		System.out.print(String.format(" %1$-12s |", dslaixe.get(vt).iNgaycong));
		System.out.print(String.format(" %1$-19s |", dslaixe.get(vt).iLuongcb));
		System.out.println(String.format(" %1$-20s |", dslaixe.get(vt).iLuong + " $"));
		System.out.println("    ----------------------------------------------------------------------------------------------------------------------------------------");

	}
	@Override
	public void luongmin() {
		// TODO Auto-generated method stub
		System.out.println("     ______________________________________________________NHÂN VIÊN CÓ LƯƠNG THẤP NHẤT_____________________________________________________");

		System.out.print(String.format("%1$8s |", "STT"));
		System.out.print(String.format(" %1$-12s |", "Mã nhân viên"));
		System.out.print(String.format(" %1$-19s |", "Họ và tên "));
		System.out.print(String.format(" %1$-15s |", "Số CMND"));
		System.out.print(String.format(" %1$-12s |", "Năm sinh"));
		System.out.print(String.format(" %1$-12s |", "Ngày công"));
		System.out.print(String.format(" %1$-19s |", "Lương cơ bản"));
		System.out.println(String.format(" %1$-20s |", "Lương chính thức" + " $"));
		System.out.println("    ---------------------------------------------------------------------------------------------------------------------------------------");
		double maxve = dslaixe.get(0).tinhluong();
		int vt = 0;
		for (int i = 0; i < dslaixe.size(); i++)
		{
			if (dslaixe.get(i).tinhluong() < maxve)
			{
				maxve = dslaixe.get(i).tinhluong();
				vt = i;
			}
		}
		System.out.print(String.format("%1$8s |", vt + 1));
		System.out.print(String.format(" %1$-12s |", dslaixe.get(vt).iManv));
		System.out.print(String.format(" %1$-19s |", dslaixe.get(vt).sHoten));
		System.out.print(String.format(" %1$-15s |", dslaixe.get(vt).iCMND));
		System.out.print(String.format(" %1$-12s |", dslaixe.get(vt).iNamsinh));
		System.out.print(String.format(" %1$-12s |", dslaixe.get(vt).iNgaycong));
		System.out.print(String.format(" %1$-19s |", dslaixe.get(vt).iLuongcb));
		System.out.println(String.format(" %1$-20s |", dslaixe.get(vt).iLuong + " $"));
		System.out.println("    --------------------------------------------------------------------------------------------------------------------------------------");

	}
}
